### ISSUE TITLE
New Rule: indent-style (tabs or spaces)

### ISSUE THREAD

**Issue by tomholford** [eslint/css#115]
### Rule details

The indent-style rule should enforce consistent indentation style in CSS files

### What type of rule is this?

Warns about a potential problem

### Example code

```css
body {
  color: black;
	background-color: white;
}
```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

The `css-indent-style` rule should enforce consistent indentation style in CSS files by checking whether tabs or spaces are used for indentation. The rule would:

1. Analyze CSS files to detect indentation characters (tabs vs. spaces)
2. Validate that the detected indentation matches the configured preference
3. Report violations when mixed indentation or incorrect indentation type is found
4. Optionally provide auto-fixing capability to convert between tabs and spaces

This rule would help teams maintain consistent code style across CSS files and prevent the common "tabs vs. spaces" debate by enforcing a team's chosen standard. It would complement existing CSS linting rules but focus specifically on the indentation character choice rather than just indentation depth.

The rule could be configurable with options like:
- `indentStyle`: "tab" or "space" (required)
- `indentSize`: Number of spaces when using space indentation (default: 2)
- `ignoreEmpty`: Whether to ignore empty lines (default: true)
- `ignoreComments`: Whether to ignore comment indentation (default: false)

**Comment by nzakas** [comment]
Yes, we're not going to be adding formatting-specific rules to this plugin. We are encouraging people to use dedicated source code formatters like Prettier or dprint to handle this instead.
